sequencer/src/api/signature_pool.rs

```diff
@@ -0,0 +1,24 @@
+use hotshot_types::light_client::StateSignature;
+use std::collections::{HashMap, VecDeque};
+
+pub(crate) const SIGNATURE_POOL_CAPACITY: usize = 100;
+
+#[derive(Debug, Default)]
+pub struct StateSignaturePool {
+    pool: HashMap<u64, StateSignature>,
+    deque: VecDeque<u64>,
```

Comment:
Can we name this something better? What is it storing?

Reply:
Come to think of it, we can combine these into `VecDeque<StateSignature>` and only store one height: the height of the first entry in the deque. Then the height of any entry `i` is `first_height + i`, and the index of any `height >= first_height` is `height - first_height`.

Reply:
Also do we want persistence for this? What should be the behavior if the node crashes and restarts? Is it ok if it loses its cached signatures?

Reply:
No persistence required. 
It's safer to just use a `HashMap` instead. 
I thought about the `VecDeque<StateSignature>` idea. The thing that I was worried about is that, we have to make sure to populate a signature for every block height.

Reply:
Also it's okay to lose all cached signatures, at least for now.

Reply:
> The thing that I was worried about is that, we have to make sure to populate a signature for every block height

Good point, I didn't think about what happens if we miss a block